File: frontend/src/lib/dto/ReadMessageDto.ts
import { MessageStatus } from "../types/enums";

// Transport shape: raw JSON from backend (dates as ISO strings)
export interface ReadMessageDto {
  id: string;
  text: string;
  userId: string;
  chatId: string;
  status: MessageStatus | number; // allow numeric flag if backend sends int
  sentAt: string; // ISO
}

File: message-service/Application/DTOs/ReadMessageDto.cs
using Domain.Enums;

namespace Application.DTOs;

public class ReadMessageDto
{
    public Guid Id { get; set; }
    public DateTime SentAt { get; set; }
    public required string Text { get; set; }
    public required Guid UserId { get; set; }
    public required Guid ChatId { get; set; }
    public required MessageStatus Status { get; set; }
}